arch/common/ramfunc.ld

```diff
@@ -7,9 +7,23 @@
 
 /* Copied from linker.ld */
 
-SECTION_DATA_PROLOGUE(.ramfunc,,)
+/* The ramfunc region must have an associated MPU entry added for it, so that
+ * code can be executed in RAM. This means the start of the ramfunc region
+ * must respect MPU alignment requirements. We include a padding section here
+ * in order to make sure the region will be placed at the correct offset
+ * in flash.
+ */
+SECTION_PROLOGUE(ramfunc_padding_section,,)
 {
 	MPU_ALIGN(__ramfunc_size);
+#ifdef CONFIG_BUILD_ALIGN_LMA
+} GROUP_DATA_LINK_IN(RAMABLE_REGION, ROMABLE_REGION)
+#else
+} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
```

Comment:
Naively, this says "If we are supposed to preserve internal alignment within sections, then this data goes into ROM, otherwise it goes into RAM".  While I buy that this creates the behavior you want to see, I really don't think this reflects a correct statement of the problem or a root cause solution. :)

And I get that the answer is "there is no data here, it's just a dummy section for padding".  But then the same question obtains: why are we padding stuff in ROM but not RAM?  And then the answer gets into non-local stuff like "because MMU" and "there's other stuff throwing off the alignment".   Then one asks "well can't we fix it where the problem is and not here?"

Reply:
This particular conditional (and `CONFIG_BUILD_ALIGN_LMA` as a whole) are a solution to a problem with `objcopy` that shows up when adjusting the LMA of an image. You can see more details on the root issue here: https://github.com/zephyrproject-rtos/zephyr/issues/58080#issuecomment-1555163828. Essentially, `objcopy` expects that the LMA of a given section will be aligned with the same restrictions as the VMA (which is a of course a strange assumption when the LMA refers to a location in ROM and VMA refers to the execution address in ram)

Essentially, if `objcopy` didn't make this assumption we would not need `CONFIG_BUILD_ALIGN_LMA` at all, but as it is this setting is needed if you want to produce a binary that `objcopy` can manipulate the LMA of

Reply:
The root cause is objcopy as outlined here and in #58080. @stephanosio proposed to fix the root cause and release the fix via Zephyr SDK. So I think this PR should be closed referring to the solution in #58080.

Reply:
I'm fine to close this PR if we can fix the root issue- I'd prefer that over patches like this one. @stephanosio, do you believe we could simply change the assumption you mention [here](https://github.com/zephyrproject-rtos/zephyr/issues/58080#issuecomment-1555163828), and fix the issue that way without causing other adverse effects?